GetBool() method

mligor opened the pull request:
Hi András, 

you made a great work here - anyway I needed reading bool from json and I implemented GetBool() method. I tried to follow your style.

If you want you can merge my changes in your repo

mligor committed:
Added GetBool() method

icza commented:
My notes:

- Your commit introduces dependencies to `encoding/json` and `strconv` packages. These could be avoided. For example my code tests for `json.Number` by checking if the type implements `interface { Float64() (float64, error) }` explicitly, to avoid having to refer to the `json` package.

- `GetXX` functions that contain the short name of a type return the type without applying built-in conversion, such as `GetInt()`. When a method uses built-in conversion, to make this obvious, the long name of the type is used (e.g. `GetInteger()`, `GetFloating()`). So this `GetBool()` should not use built-in conversion, the one that uses should be called `GetBoolean()`.

mligor commented:
Thanks for the notes. I will try to solve it - I'm in my first week using golang :)
Also code coverage should not decrease.

mligor committed:
- Removed dependencies to strconv and encoding/json
- Renamed GetBool to GetBoolean
- Improved parsing string bool values (added support for Yes/No strings)
- Extended task cases to have 100% covarage

icza commented:
Accepted the PR. Made some subsequent commits to improve docs and consistency within the library.
